lib/ash_authentication/secret.ex

```diff
@@ -124,10 +125,38 @@ defmodule AshAuthentication.Secret do
 
   @doc false
   def secret_for(module, secret_name, resource, opts, context) do
-    if module.__secret_for_arity__() == 4 do
-      module.secret_for(secret_name, resource, opts, context)
-    else
-      module.secret_for(secret_name, resource, opts)
+    result =
+      if module.__secret_for_arity__() == 4 do
+        module.secret_for(secret_name, resource, opts, context)
+      else
+        module.secret_for(secret_name, resource, opts)
+      end
+
+    case result do
+      {:ok, secret}
+      when is_binary(secret) or is_list(secret) or is_map(secret) or is_boolean(secret) ->
```

Comment:
I don't really understand why a secret should be any one of these particular types? I think we should just raise when the secret doesn't return `ok/error` tuple and let secrets return whatever they want.